x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.test.ts

```diff
@@ -0,0 +1,512 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ActionTypeRegistry, ActionTypeRegistryOpts } from './action_type_registry';
+import { ActionsClient } from './actions_client';
+import { ExecutorType } from './types';
+import { ActionExecutor, TaskRunnerFactory, ILicenseState } from './lib';
+import { taskManagerMock } from '../../task_manager/server/mocks';
+import { actionsConfigMock } from './actions_config.mock';
+import { licenseStateMock } from './lib/license_state.mock';
+import { licensingMock } from '../../licensing/server/mocks';
+import { httpServerMock } from '../../../../src/core/server/mocks';
+import { auditServiceMock } from '../../security/server/audit/index.mock';
+import { usageCountersServiceMock } from 'src/plugins/usage_collection/server/usage_counters/usage_counters_service.mock';
+
+import {
+  elasticsearchServiceMock,
+  savedObjectsClientMock,
+} from '../../../../src/core/server/mocks';
+import { actionExecutorMock } from './lib/action_executor.mock';
+import { ActionsAuthorization } from './authorization/actions_authorization';
+import { actionsAuthorizationMock } from './authorization/actions_authorization.mock';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { elasticsearchClientMock } from '../../../../src/core/server/elasticsearch/client/mocks';
+
+jest.mock('../../../../src/core/server/saved_objects/service/lib/utils', () => ({
+  SavedObjectsUtils: {
+    generateId: () => 'mock-saved-object-id',
+  },
+}));
+
+jest.mock('./lib/track_legacy_rbac_exemption', () => ({
+  trackLegacyRBACExemption: jest.fn(),
+}));
+
+jest.mock('./authorization/get_authorization_mode_by_source', () => {
+  return {
+    getAuthorizationModeBySource: jest.fn(() => {
+      return 1;
+    }),
+    AuthorizationMode: {
+      Legacy: 0,
+      RBAC: 1,
+    },
+  };
+});
+
+const defaultKibanaIndex = '.kibana';
+const unsecuredSavedObjectsClient = savedObjectsClientMock.create();
+const scopedClusterClient = elasticsearchServiceMock.createScopedClusterClient();
+const actionExecutor = actionExecutorMock.create();
+const authorization = actionsAuthorizationMock.create();
+const executionEnqueuer = jest.fn();
+const ephemeralExecutionEnqueuer = jest.fn();
+const request = httpServerMock.createKibanaRequest();
+const auditLogger = auditServiceMock.create().asScoped(request);
+const mockUsageCountersSetup = usageCountersServiceMock.createSetupContract();
+const mockUsageCounter = mockUsageCountersSetup.createUsageCounter('test');
+
+const mockTaskManager = taskManagerMock.createSetup();
+
+let actionsClient: ActionsClient;
```

Comment:
These tests look like they need to be updated for `ConnectorTokenClient` :)

Reply:
yep